tests/acceptance/test_external_studies.py

```diff
@@ -64,8 +65,22 @@
 STUDY_DIR = Path(__file__).parent / "external_studies"
 
 
+@pytest.fixture
+def reset_module_namespace():
+    """
+    Studies often use the same names for modules (e.g. codelists.py, variables_lib.py)
+    Ensure that we clean up the module namespace after each external study test.
```

Comment:
Oh, of course. I should have realised this would be problem. Your workaround looks like it will do the trick. I suppose there are other side-effects that dataset definitions could have (mucking about with `sys.path`, for instance) which might end up causing us problems down the line. But let's worry about those when they arise.

The "proper" way to do this would be to in a subprocess where none of this stuff can leak back to the caller. But I don't think it's worth changing anything until we hit actual problems.